controller/src/resource_controller/action.rs

```diff
@@ -151,8 +182,52 @@ async fn creation_completed_action(r: &ResourceInterface) -> Result<CreationActi
     }
 }
 
+/// Destruction for a resource may be required if the resources `DestructionPolicy` is
+/// `OnTestSuccess` or `OnTestCompletion`.
+async fn deletion_needed(r: &ResourceInterface) -> Result<bool> {
```

Comment:
nit: we already have `is_deletion_requested`, so maybe
```suggestion
async fn is_deletion_required(r: &ResourceInterface) -> Result<bool> {
```